applications/nrf_desktop/src/modules/ble_scan.c

```diff
@@ -277,8 +272,110 @@ static bool le_param_req(struct bt_conn *conn, struct bt_le_conn_param *param)
 	return true;
 }
 
+static int store_subscribed_peers(void)
+{
+	if (IS_ENABLED(CONFIG_SETTINGS)) {
+		char key[] = MODULE_NAME "/" SUBSCRIBED_PEERS_STORAGE_NAME;
+		int err = settings_save_one(key, subscribed_peers,
+					    sizeof(subscribed_peers));
+
+		if (err) {
+			LOG_ERR("Problem storing subscribed_peers (err %d)",
+				err);
+			return err;
+		}
+	}
+
+	return 0;
+}
+
+static void reset_subscribers(void)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(subscribed_peers); i++) {
+		bt_addr_le_copy(&subscribed_peers[i].addr, BT_ADDR_LE_NONE);
+		subscribed_peers[i].peer_type = PEER_TYPE_COUNT;
+	}
+}
+
+static int settings_set(const char *key, size_t len_rd,
+			settings_read_cb read_cb, void *cb_arg)
+{
+	if (!strcmp(key, SUBSCRIBED_PEERS_STORAGE_NAME)) {
+		ssize_t len = read_cb(cb_arg, &subscribed_peers,
+				      sizeof(subscribed_peers));
+		if (len != sizeof(subscribed_peers)) {
+			LOG_ERR("Can't read subscribed_peers from storage");
+			module_set_state(MODULE_STATE_ERROR);
+			return len;
+		}
+	}
+
+	return 0;
+}
+
+static void verify_bond(const struct bt_bond_info *info, void *user_data)
+{
+	size_t i;
+
+	for (i = 0; i < ARRAY_SIZE(subscribed_peers); i++) {
+		if (!bt_addr_le_cmp(&subscribed_peers[i].addr, &info->addr)) {
+			break;
```

Comment:
return

Reply:
then `size_t i` can go to for and condition check from line 326 can go away

Reply:
ok